skyportal/handlers/api/spatial_catalog.py

```diff
@@ -0,0 +1,379 @@
+        parameters:
+          - in: path
+            name: catalog_id
+            required: true
+            schema:
+              type: integer
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+        with self.Session() as session:
+            stmt = SpatialCatalog.select(session.user_or_token, mode="delete").where(
+                SpatialCatalog.id == int(catalog_id)
+            )
+            catalog = session.scalars(stmt).first()
+            if catalog is None:
+                return self.error(f'Missing catalog with ID {catalog_id}')
+
+            IOLoop.current().run_in_executor(None, lambda: delete_catalog(catalog.id))
+
+            self.push_all(action="skyportal/REFRESH_SPATIAL_CATALOGS")
+            return self.success()
+
+
+class SpatialCatalogASCIIFileHandler(BaseHandler):
+    @permissions(['Upload data'])
+    def post(self):
+        """
+        ---
+        description: Upload spatial catalog from ASCII file
+        tags:
+          - galaxys
+        requestBody:
+          content:
+            application/json:
+              schema: SpatialCatalogASCIIFileHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New spatial catalog ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        json = self.get_json()
+        catalog_data = json.pop('catalogData', None)
+        catalog_name = json.pop('catalogName', None)
+
+        if catalog_data is None:
+            return self.error(message="Missing catalog_data")
+
+        try:
+            catalog_data = pd.read_table(StringIO(catalog_data), sep=",").to_dict(
+                orient='list'
+            )
+        except Exception as e:
+            return self.error(f"Unable to read in galaxy file: {e}")
+
+        if catalog_name is None:
+            return self.error("catalog_name is a required parameter.")
+        if catalog_data is None:
+            return self.error("catalog_data is a required parameter.")
+
+        # check RA bounds
+        if any([(x < 0) or (x >= 360) for x in catalog_data['ra']]):
+            return self.error("ra should span 0=<ra<360.")
+
+        # check Declination bounds
+        if any([(x > 90) or (x < -90) for x in catalog_data['dec']]):
+            return self.error("declination should span -90<dec<90.")
```

Comment:
Same comment here, please check that the data is indeed a dict and that it does have `ra` and `dec` keys ( you can also do some data type checking for other variables, but this is the obvious one that could cause problems pretty easily.

Reply:
added.